File: web-app/models/enums/TaskStatus.ts
export enum TaskStatus {
  Todo,
  InProgress,
  Completed
}

export class TaskStatusMapper {
  readonly Description = Object.freeze({
    [TaskStatus.Todo]: "To do",
    [TaskStatus.InProgress]: "In progress",
    [TaskStatus.Completed]: "Completed"
  });

  readonly Color = Object.freeze({
    [TaskStatus.Todo]: "primary",
    [TaskStatus.InProgress]: "amber-darken-4",
    [TaskStatus.Completed]: "green-darken-2"
  });

  getDescription(status: TaskStatus): string {
    return this.Description[status];
  }

  getColour(status: TaskStatus): string {
    return this.Color[status];
  }
}

File: app/app/src/main/java/com/solutions/note_it/data/TaskStatus.kt
package com.solutions.note_it.data

import androidx.compose.ui.graphics.Color

enum class TaskStatus(val desc: String, val color: Color) {
    Todo("To do", Color.Gray),
    InProgress("In progress", Color.Yellow),
    Completed("Completed", Color.Green),
    Archived("Archived", Color.LightGray)
}